lib/perl5db.pl

```diff
@@ -2786,7 +2786,7 @@ sub _cmd_l_main {
     # If the line is '$something', assume this is a scalar containing a
     # line number.
     # Set up for DB::eval() - evaluate in *user* context.
-    if ( my ($var_name) = $spec =~ /\A(\$.*)/s ) {
+    if ( my ($var_name) = $spec =~ /\A(\$[\w]+)\z/s ) {
```

Comment:
I didn't change that because it wasn't critical to getting the patch working; I'd rather address style fixes as a separate patch. Agreed on the `[\w]+`, will change that, and the more-strict regex is definitely better.

Reply:
Going back, you're right, that's just silly and redundant. Removing `$var_name`.